internal/infrastructure/kubernetes/ratelimit/resource_provider_test.go

```diff
@@ -36,6 +36,32 @@ var ownerReferenceUID = map[string]types.UID{
 	ResourceKindServiceAccount: "test-owner-reference-uid-for-service-account",
 }
 
+func TestRateLimitLabelSelector(t *testing.T) {
+
+	cases := []struct {
+		name     string
+		expected []string
+	}{
+		{
+			name: "rateLimit-labelSelector",
+			expected: []string{
+				"app.kubernetes.io/name=envoy-ratelimit",
+				"app.kubernetes.io/component=ratelimit",
+				"app.kubernetes.io/managed-by=envoy-gateway",
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			got := LabelSelector()
+			require.Equal(t, tc.expected, got)
```

Comment:
better go with `require.ElementsMatch`, since we only can promise the values of `got` as expected not the orders.

Reply:
Okay, do I only need to change these two things?

Reply:
let's give it a try now

Reply:
It seems to be okay this time 😃